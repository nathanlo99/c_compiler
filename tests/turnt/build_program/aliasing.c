
int wain(int a, int b) {
  int *c = NULL;
  c = &a;

  a = 69; // Seemingly unused assignment
  b = *c; // Sneakily used here
  a = 70;

  return a + b; // 69 + 70 = 139
}
