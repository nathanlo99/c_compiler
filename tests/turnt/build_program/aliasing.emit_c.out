int wain(int a, int b) {
  int* c = NULL;
  c = &a;
  a = 69;
  b = *c;
  a = 70;
  return (a + b);
}

